fgpyo/alignment/__init__.py

```diff
@@ -0,0 +1,232 @@
+                    query_buffer.append(q)
+                    target_buffer.append(t)
+
+                    if cig.operator == CigarOp.EQ:
+                        align_buffer.append(match_char)
+                    elif cig.operator == CigarOp.X:
+                        align_buffer.append(mismatch_char)
+                    elif cig.operator == CigarOp.M:
+                        align_buffer.append(match_char if q == t else mismatch_char)
+                    else:
+                        raise ValueError(f"Unsupported cigar operator: {cig.operator}")
+
+                    q_offset += 1
+                    t_offset += 1
+
+        return ["".join(query_buffer), "".join(align_buffer), "".join(target_buffer)]
+
+    def sub_by_query(self, start: int, end: int) -> "Alignment":
+        """Returns a subset of the Alignment representing the region defined by start
+        and end on the query sequence. The returned alignment will contain the entire
+        query and target sequences, but will have adjusted query_start and target_start
+        positions and an updated cigar. The score is set to 0.
+
+        Args:
+            start (int): The 1-based inclusive position of the first base on the query
+                sequence to include.
+            end (int): The 1-based inclusive position of the last base on the query
+                sequence to include.
+
+        Returns:
+            A new Alignment with updated coordinates and cigar.
+        """
+        if not (
+            self.query_start <= start <= self.query_end
+            and self.query_start <= end <= self.query_end
+        ):
+            raise ValueError("start or end is outside of aligned region of target sequence")
+
+        return self.sub(start, end, self.query_start, lambda elem: elem.operator.consumes_query)
+
+    def sub_by_target(self, start: int, end: int) -> "Alignment":
+        """Returns a subset of the Alignment representing the region defined by start and end on
+        the target sequence. The returned alignment will contain the entire query and target
+        sequences, but will have adjusted query_start and target_start positions and an updated
+        cigar. The score is set to 0.
+
+        Args:
+            start (int): The 1-based inclusive position of the first base on the target sequence
+                to include.
+            end (int): The 1-based inclusive position of the last base on the target sequence
+                to include.
+
+        Returns:
+            A new Alignment with updated coordinates and cigar.
+        """
+        if not (
+            self.target_start <= start <= self.target_end
+            and self.target_start <= end <= self.target_end
+        ):
+            raise ValueError("start or end is outside of aligned region of target sequence")
+
+        return self.sub(
+            start, end, self.target_start, lambda elem: elem.operator.consumes_reference
+        )
+
+    def sub(
```

Comment:
Make "private" as the docstring suggest
```suggestion
    def _sub(
```